File: api/src/entity/Account.ts
import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm';
import { Upload } from './Upload';

@Entity()
export class Account {

  @PrimaryGeneratedColumn()
  id: number;

  @Column({
    unique: true
  })
  name: string;

  @Column()
  dateHeader: string;

  @Column()
  descriptionHeader: string;

  @Column()
  amountHeader: string;

  @Column('double precision', { default: 0, scale: 2 })
  startingAmount: number;

  @Column('double precision', { default: 0, scale: 2 })
  balance: number;

  @Column({ default: false })
  amountsInverted: boolean;

  @Column({
    default: '#999999'
  })
  color: string;

  @OneToMany(() => Upload, upload => upload.account)
  uploads: Upload[];

}

File: backend/model/account.go
package model

import (
	"gorm.io/gorm"
)

type Account struct {
	gorm.Model
	Name              string `gorm:"unique;not null;default:null"`
	DateHeader        string `gorm:"not null;default:null"`
	DescriptionHeader string `gorm:"not null;default:null"`
	AmountExpression  string `gorm:"not null;default:null"`
	StartingAmount    int64  `gorm:"not null;default:0"`
	Balance           int64  `gorm:"not null;default:0"`
	Color             *string
	LoginURL          *string
	Uploads           []Upload
}
